components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/mqtt/MqttListener.java

```diff
@@ -0,0 +1,272 @@
+            this.sequential =
+                    Boolean.parseBoolean(mqttProperties.getProperty(PollingConstants.INBOUND_ENDPOINT_SEQUENTIAL));
+        }
+
+        confac = new MqttConnectionFactory(mqttProperties);
+        contentType = confac.getContent();
+
+        injectHandler =
+                new MqttInjectHandler(injectingSeq, onErrorSeq, sequential,
+                        synapseEnvironment, contentType);
+
+        if (mqttProperties.getProperty(MqttConstants.MQTT_USERNAME) != null) {
+            userName = mqttProperties.getProperty(MqttConstants.MQTT_USERNAME);
+        }
+
+        if (mqttProperties.getProperty(MqttConstants.MQTT_PASSWORD) != null) {
+            password = mqttProperties.getProperty(MqttConstants.MQTT_PASSWORD);
+        }
+
+        cleanSession =
+                Boolean.parseBoolean(mqttProperties.getProperty(MqttConstants.MQTT_SESSION_CLEAN));
+        mqttBlockingSenderEnable =
+                Boolean.parseBoolean(mqttProperties.getProperty(MqttConstants.MQTT_BLOCKING_SENDER));
+
+    }
+
+    public InboundRequestProcessorFactoryImpl getListner() {
+        return listner;
+    }
+
+    @Override
+    public void destroy() {
+
+        try {
+            if (mqttAsyncCallback != null) {
+                mqttAsyncCallback.disconnect();
+            } else
+                mqttClient.disconnect();
+            log.info("Disconnected.");
+        } catch (MqttException e) {
+            log.error("Error while disconnecting from the remote server...", e);
+        }
+
+    }
+
+    @Override
+    public void init() {
+
+        log.info("MQTT inbound endpoint: " + name + " initializing ...");
+
+        if (mqttBlockingSenderEnable)
+            initSyncClient();
+        else
+            initAsyncClient();
+    }
+
+    /**
+     * MQTT synchrounous client
+     */
+    public void initSyncClient() {
+        mqttClient = confac.getMqttClient();
+
+        MqttSyncCallback mqttSyncCallback = new MqttSyncCallback(injectHandler);//changed
+        mqttClient.setCallback(mqttSyncCallback);
+
+        if (confac != null) {
+            mqttSyncCallback.setMqttSyncCallback(confac);
+        }
+        if (params != null) {
+            mqttSyncCallback.setParams(params);
+        }
+
+        MqttConnectOptions opt = new MqttConnectOptions();
+        opt.setCleanSession(cleanSession);
+        if (opt != null) {
+            mqttSyncCallback.setOpt(opt);
+        }
+
+        if (userName != null && password != null) {
+            opt.setUserName(userName);
+            opt.setPassword(password.toCharArray());
+        }
+
+        try {
+            mqttClient.connect();
+            if (confac.getTopic() != null) {
+                mqttClient.subscribe(confac.getTopic());
+            }
+            log.info("Connected to the remote server.");
+        } catch (MqttException e) {
+            if (!mqttClient.isConnected()) {
```

Comment:
Make this retry interval as a parameter and optional. eg: when not specified default to 1000ms